parsing/parser.mly

```diff
@@ -418,6 +418,21 @@ let wrap_sig_ext ~loc body ext =
 let wrap_mksig_ext ~loc (item, ext) =
   wrap_sig_ext ~loc (mksig ~loc item) ext
 
+let mk_quotedext ~loc ~shift (id, content, strloc, delim) =
```

Comment:
It's late and my head hurts looking at this code. Why is it so complex if you did all the work in the lexer?

Reply:
Because life is not all pink and sometimes you need to get your hand dirty with locations. :)

The complex part of this code computes the position of `foo.bar` in `{%foo.bar-baz|...baz|}`. It's only complicated because the representation of position is mildly akward.

Moving  this code to the lexer is feasible, but requires differentiating code path between strings and quoted extensions. I do not see how that improves anything.

Reply:
One thing that I find suspicious for example, is that the values passed to the `~shift` parameter depend on the lexical representation of the token. This sounds like lexer-level logic rather than parser-level logic.

At least it would be nice there were comments on what various parts of the code do. (What is the `shift` argument for? What is the difference between `loc` and `strloc`?)

Reply:
The last commit moves it to the lexer.